### ISSUE TITLE
Feature Request: Wireguard and Firewall

### ISSUE THREAD

**Issue by 8itlew7r** [celzero/rethink-app#976]
When a Wireguard is disabled, a kill-switch must be enabled that prevents apps from connecting when the Wireguard has been deactivated or has difficulty connecting to its endpoint.

Firewall rules can be tailored to specific applications instead of being universal, allowing for greater flexibility without compromising security.

Thank you for your hard work!

**Comment by ignoramous** [comment]
Makes sense. The UI for this is tricky to implement and so we left this out from the first version. Let's see. Thanks.

**Comment by ignoramous** [comment]
Related:
- #1014

**Comment by ignoramous** [comment]
In [`v055b`](https://github.com/celzero/rethink-app/releases/tag/v0.5.5b) we've implemented "Lockdown" for WireGuard.

**Comment by maltfield** [comment]
@ignoramous I think this issue still isn't fixed?

To my horror, I recently discovered that my IP address was leaked when using this app. That could be a death sentence for some folks.

Here's the problem: the "lockdown" option is set in the individual wireguard config. We need the lockdown option to be set in the general app's "proxy" settings.

Recently I was trying to watch a video on YouTube, which has been nearly unusable for most VPN exits for about a year. To get it working, I have to cycle through VPN exits until I finally get an IP that isn't banned by YouTube. This typically requires me to switch between 5-20 WireGuard configs.

So the process is:

1. Open "Rethink" app
2. Press "START" 
3. Click Proxy
4. Click the toggle to enable one of the many wireguard configs
5. Go into the WireGuard config's settings and enable the toggle on "Lockdown" and "Always-on"
6. Check YouTube. If the IP is banned, then go into the enabled Wireguard config's settings and disable "Lockdown" and disable "Always-on"
7. Go back, and click the toggle to disable the wireguard config
8. GOTO 4

This process is *extremely* problematic, because it means that the user is left vulnerable (exposing their IP address!) at many parts:

1. The user is vulnerable after step number 2 above, when they click "START" but before they've enabled the WireGuard proxy
2. The user is vulnerable after step number 7 above, when they've disabled the WireGuard proxy

This is an issue for two reasons:

1. The user cannot toggle-enable a given WireGuard config without first clicking "START" (I created a [bug report](https://github.com/celzero/rethink-app/issues/2416) to fix this)
2. The user cannot disable a given WireGuard config without first disabling "Lockdown" mode (I created another [bug report](https://github.com/celzero/rethink-app/issues/2417) to fix this too)

**Comment by onetimecontributor** [comment]
(disclaimer) I'm not affiliated with rdns in any way

"That could be a death sentence for some folks." don't stop, exaggerate even further, "death to humanity". 

Though rdns is great, it has never promised something that's addressed by Tor.

**Comment by ignoramous** [comment]
> @ignoramous I think this issue still isn't fixed?

It isn't implemented for *other* proxies (HTTP/SOCKS5), it has been implemented for WireGuard. In fact, v055u should be much better in dealing with *Lockdown*.

> I have to cycle through VPN exits until I finally get an IP that isn't banned by YouTube. This typically requires me to switch between 5-20 WireGuard configs.

Gotcha, but this sounds like a super specific usecase not many would have?

> The user is vulnerable after step number 2 above, when they click "START" but before they've enabled the WireGuard proxy ... The user is vulnerable after step number 7 above, when they've disabled the WireGuard proxy

Please track:
- https://github.com/celzero/rethink-app/issues/2417
